Realm/RLMSyncConfiguration.h

```diff
@@ -39,6 +39,20 @@ NS_ASSUME_NONNULL_BEGIN
  */
 @property (nonatomic, readonly) NSURL *realmURL;
 
+/**
+ A local path to a file containing the trust anchors for SSL connections.
+
+ Only the certificates stored in the PEM file (or any certificates signed by it,
+ if the file contains a CA cert) will be accepted when initiating a connection
+ to a server. This prevents certain certain kinds of man-in-the-middle (MITM)
+ attacks, and can also be used to trust a self-signed certificate which would
+ otherwise be untrusted.
+
+ On macOS, the file may be in any of the formats supported by SecItemImport(),
+ including PEM and .cer (see SecExternalFormat for a complete list of possible
+ formats). On iOS and other platforms, only DER .cer files are supported.
+ */
+@property (nonatomic, nullable) NSURL *pinnedCertificateURL;
```

Comment:
Is there a reason you can set this property? Since any certificate validation should be enabled for both auth calls and sync, it seems a bit odd you can set it on both the `SyncManager` and the `SyncConfiguration` as it just opens up for mismatching config settings. Shouldn't the configuration just inherit it from the `SyncManager` and then make this readonly?

Reply:
This is automatically set to the one used for auth when getting a configuration from a user, but it's possible to have the sync server and auth servers on different domains, which requires different pinned certificates. `RLMSyncConfiguration` should probably be entirely immutable like the Swift version is, but for now it has mutable fields.